iaas/digitalocean/digitalocean.go

```diff
@@ -1,465 +1,118 @@
+		SSHKeyID    int    `json:"SSHKeyID"`
+	} `json:"Driver"`
 }
 
-// GetSSHPrivateKeyPath the path may change according to the
-// environment variable GOFM_SSH_PRIVATEKEY_PATH or by using
-// the SetSSHPrivateKeyPath
-func (do *Digitalocean) GetSSHPrivateKeyPath() (path string) {
-	path = os.Getenv("GOFN_SSH_PRIVATEKEY_PATH")
-	if path != "" {
-		return
-	}
-	path = do.sshPrivateKeyPath
-	if path != "" {
+func getConfig(machineDir, hostName string) (config *driverConfig, err error) {
+	configPath := fmt.Sprintf("%s/%s/config.json", machineDir, hostName)
+	raw, err := ioutil.ReadFile(configPath)
+	if err != nil {
 		return
 	}
-	do.sshPrivateKeyPath = filepath.Join(gofnssh.KeysDir, gofnssh.PrivateKeyName)
-	path = do.sshPrivateKeyPath
-	return
-}
-
-// GetSnapshotName returns snapshot name or default if empty
-func (do Digitalocean) GetSnapshotName() string {
-	if do.SnapshotName == "" {
-		return defaultSnapshotName
-	}
-	return do.SnapshotName
-}
-
-// GetRegion returns region or default if empty
-func (do Digitalocean) GetRegion() string {
-	if do.Region == "" {
-		return defaultRegion
-	}
-	return do.Region
-}
-
-// GetSize returns size or default if empty
-func (do Digitalocean) GetSize() string {
-	if do.Size == "" {
-		return defaultSize
-	}
-	return do.Size
-}
-
-// GetImageSlug returns image slug  or default if empty
-func (do Digitalocean) GetImageSlug() string {
-	if do.ImageSlug == "" {
-		return defaultImageSlug
-	}
-	return do.ImageSlug
-}
-
-// Auth in Digitalocean API
-func (do *Digitalocean) Auth() (err error) {
-	apiURL := os.Getenv("DIGITALOCEAN_API_URL")
-	key := os.Getenv("DIGITALOCEAN_API_KEY")
-	if key == "" {
-		err = errors.New("You must provide a Digital Ocean API Key")
+	err = json.Unmarshal(raw, &config)
+	if err != nil {
 		return
 	}
-	tokenSource := oauth2.StaticTokenSource(&oauth2.Token{
-		AccessToken: key,
-	})
-	oauthClient := oauth2.NewClient(context.Background(), tokenSource)
-	do.client = godo.NewClient(oauthClient)
-	if apiURL != "" {
-		do.client.BaseURL, err = url.Parse(apiURL)
-		if err != nil {
-			return
-		}
-	}
 	return
 }
 
-// CreateMachine on digitalocean
-func (do *Digitalocean) CreateMachine() (machine *iaas.Machine, err error) {
-	err = do.Auth()
-	if err != nil {
-		return
-	}
-	lo := godo.ListOptions{
-		Page:    1,
-		PerPage: 999999,
-	}
-	snapshots, _, err := do.client.Snapshots.List(do.Ctx, &lo)
-	if err != nil {
-		return
-	}
-	snapshot := godo.Snapshot{}
-	for _, s := range snapshots {
-		if s.Name == do.GetSnapshotName() {
-			snapshot = s
-			break
-		}
-	}
-	if snapshot.Name == "" && do.ErrorIfSnapshotNotExist {
-		err = ErrSnapshotNotFound
-		return
-	}
-	image := godo.DropletCreateImage{
-		Slug: do.GetImageSlug(),
-	}
-	if snapshot.Name != "" {
-		id, _ := strconv.Atoi(snapshot.ID)
-		image = godo.DropletCreateImage{
-			ID: id,
-		}
-	}
-	sshKey, err := do.getSSHKeyForDroplet()
-	if err != nil {
-		return
-	}
+func New(token string) (do *provider, err error) {
```

Comment:
exported func New returns unexported type *digitalocean.provider, which can be annoying to use